### PR TITLE
[stdlib] Reinstate some tests

### PR DESCRIPTION
**Pull request by benrimmington**
Follow-up to: apple/swift#23423

I've tested (on macOS 11.2.1 and Xcode 12.4) with:

```shell
utils/run-test \
  --build-dir ../build/Ninja-RelWithDebInfoAssert \
  --verbose \
  test/stdlib/CodableTests.swift \
  test/stdlib/FloatingPoint.swift.gyb \
  test/stdlib/SetTrapsObjC.swift \
  validation-test/stdlib/DictionaryTrapsObjC.swift
```

I've replaced multi-line [substitutions][] with one of:

```swift
// RUN: %target-run-simple-swift(...)
// RUN: %target-run-simple-swiftgyb(...)
```

I've marked the remaining test failures with one of:

```swift
// FIXME: <rdar://problem/49026133>
// FIXME: <rdar://problem/48917925>
// FIXME: <https://bugs.swift.org/browse/SR-14282>
```

[substitutions]: <https://github.com/apple/swift/blob/main/docs/Testing.md#substitutions-in-lit-tests>

### CODE REVIEW

**Comment by benrimmington** [comment]
@swift-ci Please test

**Comment by benrimmington** [comment]
@swift-ci Please test Windows platform

**Comment by benrimmington** [comment]
```
Failed Tests (1):
  Swift(watchsimulator-i386) :: IRGen/marker_protocol.swift
```

**Comment by benrimmington** [comment]
@swift-ci Please test macOS platform

**Comment by benrimmington** [comment]
@swift-ci Please test Windows platform

**Comment by benrimmington** [comment]
```
Failed Tests (2):
  Swift(windows-x86_64) :: SILOptimizer/definite-init-wrongscope.swift
  Swift(windows-x86_64) :: IRGen/marker_protocol.swift
```

**Comment by benrimmington** [comment]
@swift-ci Please test Windows platform

File: validation-test/stdlib/DictionaryTrapsObjC.swift [hunk 1]
```diff
@@ -205,6 +198,10 @@ DictionaryTraps.test("ForcedNonverbatimBridge.Value")
   _ = d1 as! Dictionary<NSObject, String>
 }
 
+// FIXME: <rdar://problem/49026133>
+// Segmentation fault: 11 (when compiling with optimizations).
+// While running pass #0 SILModuleTransform "SILGenCleanup".
```
Comment (lorentey) [inline validation-test/stdlib/DictionaryTrapsObjC.swift:203]: Hm frustratingly, this compiler crash is a new issue that is unrelated to the original problem (which was a runtime abort() call in the `*_debug.sanity` tests). Serves us right for leaving this test disabled for so long... 😞

If you still have a crash log handy, could you please file a bug for it?

File: test/stdlib/FloatingPoint.swift.gyb [hunk 2]
```diff
@@ -827,10 +822,11 @@ FloatingPoint.test("${FloatSelf}/{Comparable,Hashable,Equatable}") {
       //  If either lhs or rhs is signaling, or if both are quiet NaNs, the
       //  result is a quiet NaN.
       if lhs.isSignalingNaN || rhs.isSignalingNaN || (lhs.isNaN && rhs.isNaN) {
-        expectTrue(min.isQuietNaN)
-        expectTrue(max.isQuietNaN)
-        expectTrue(minMag.isQuietNaN)
-        expectTrue(maxMag.isQuietNaN)
+        // FIXME: <rdar://problem/49026133>
```
Comment (lorentey) [inline test/stdlib/FloatingPoint.swift.gyb:825]: Cc @stephentyrone -- He has been tracking this.

File: test/stdlib/CodableTests.swift [hunk 3]
```diff
@@ -1,15 +1,6 @@
-// Copyright (c) 2014 - 2019 Apple Inc. and the Swift project authors
-// Licensed under Apache License v2.0 with Runtime Library Exception
-//
-// See https://swift.org/LICENSE.txt for license information
-// See https://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
-//
-//===----------------------------------------------------------------------===//
-//
```
Comment (lorentey) [inline test/stdlib/CodableTests.swift:8]: Why remove the copyright header?

**Review by lorentey — APPROVED** [review]
This is excellent, thank you so much for working on this!

File: test/stdlib/CodableTests.swift (hunk 3 above)
Comment (benrimmington) [reply test/stdlib/CodableTests.swift:8]: I've restored the copyright header, with the correct years.
(However, the majority of stdlib tests don't have copyright headers.)

File: validation-test/stdlib/DictionaryTrapsObjC.swift (hunk 1 above)
Comment (benrimmington) [reply validation-test/stdlib/DictionaryTrapsObjC.swift:203]: > If you still have a crash log handy, could you please file a bug for it?

I've filed SR-14282.

**Comment by benrimmington** [comment]
@swift-ci Please test
